### ISSUE TITLE
[Suggestion] Add a count for Move/Index?

### ISSUE THREAD

**Issue by nowjon** [photoprism/photoprism#639]
Hey, 

I wanted to suggest a count for how many pictures are done or how many are left when Moving or Indexing. Currently, it's very hard to discern how long it will take to import the items. 

Thanks!

**Comment by lastzero** [comment]
Would require a two pass scan. Also processing RAWs for example takes much longer than indexing small JPEGs. Making a predicting here is more art than science, so we left this todo open for later.

**Comment by erahhal** [comment]
Perhaps a parallel dynamic progress would make it more responsive, e.g. two threads, one crawling the directory tree and updating the model for how long it should take, while another is doing the indexing, so no need to hold up indexing with two serial passes.  People are used to seeing progress and estimated time change significantly towards the beginning of a job.

**Comment by lastzero** [comment]
You could use the average time per file for estimates. Just need to query the files table. Takes time until we can eventually implement this in the UI. Feature requests growing faster than our budget.

**Comment by JohannesRudolph** [comment]
Would a minimal version that just shows the number of files/MiBs left to process in the import dir (as opposed to a time estimate) be a reasonable way to cut the scope? Alternatively show some aggregate stats about the "current import batch" like how long has the importer been running for, how many images/MiBs has it processed already as this gives some user feedback on progress as well.

**Comment by lastzero** [comment]
It would still be a two-pass scan, checking all folders. Depending on the file system and disk performance, this could cause unwanted side effects for some users, so it probably won't be done and tested in 10 minutes. We will be happy to revisit this once the multi-user and in-app upgrades are ready.

**Comment by javixuwi** [comment]
maybe we can start showing the number of indexed files slash total files. The time is important, but more important some feedback of what is going on at the server.

**Comment by lastzero** [comment]
That would require a two pass scan, meaning it will be slower. Also it's possible that files or directories are added after indexing has started, meaning the number of files would be wrong in this case.

**Comment by RedShiftLight** [comment]
This sounds like a perfect opportunity for a switch to turn on the proposed feature. If you don't mind a 2 pass scan, then feel free to enable it. Else, no progress bar. I thought about the suggestion to count in 1 thread while indexing with other threads and was in favor of it until I thought about the read/(and possibly write) operations happening on the disk. I agree with @lastzero in that I think it would be simpler and faster overall to do counting before processing. In the space for the progress bar, you can put a cynical message saying "Counting files to create your damn progress bar!".

I have about 400GB of images and movies that spans 23 years of family photos, so I'd expect it to take a long time to get a count of non-indexed files. The wife takes far too many pictures :) Windows takes a long time to count and get disk usage for my folder. I'd hate to see how long my Raspberry Pi takes to do it. I am currently indexing on a laptop so I can copy the cache onto my Pi once complete. I expect it will take a few days.

As a secondary feature, it would be nice to see a indexing "mini-log" presented on the screen showing a timestamp for each time indexing started, a count of the number of files indexed, and a finish timestamp. While it is running, you can update the count every 30 seconds. And add a button to clear the log if we wanted.

I love your product and if I decide to keep using it (I'm evaluating it now) I plan to make a donation. Keep up the good work!

**Comment by TrySpace** [comment]
I agree there should be a better indication of progress. The only clue I have right now (first time run) is I hear my harddrive's infernal noise. This wouldn't be the case if it were and SSD.

**Comment by anthraxn8b** [comment]
While this would be a great feature (I currently watch photoprism's docker stats CPU%... and hope that it will be over soon...), this information could also be provided at a prometheus metric endpoint... Maybe even in the first step. Second step could be the integration into the UI...

**Comment by gshaowei6** [comment]
I opened PR #5586 with a first-step implementation based on the Prometheus metrics suggestion above: it tracks whether import/index is running, the number of files processed in the current or most recent run, processed bytes, and start/finish timestamps. It intentionally avoids a two-pass scan, total count, or ETA so it should not add extra filesystem traversal overhead.
